web3-react/src/libs/hooks.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react'
+import { getConnection, ConnectionType } from './connections'
+import { Connector } from '@web3-react/types'
+
+export const useEagerlyConnect = () => {
+  useEffect(() => {
+    connect(getConnection(ConnectionType.NETWORK).connector)
+    connect(getConnection(ConnectionType.GNOSIS_SAFE).connector)
+  }, [])
+}
+
+async function connect(connector: Connector) {
+  try {
+    if (connector.connectEagerly) {
+      await connector.connectEagerly()
```

Comment:
I believe this is always defined in `web3-react`, at least looking at the type?

Reply:
`deactivate?(...args: unknown[]): Promise<void> | void;` does not seem like it always defined 🤔

Reply:
Hm, this one is correct, however, I was referring to `connectEagerly`. I wonder why ESLint didn't pick that up.

Reply:
Same story here:
` connectEagerly?(...args: unknown[]): Promise<void> | void;`